staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -71,6 +85,8 @@ func (a *Authenticator) AuthenticateRequest(req *http.Request) (user.Info, bool,
 		}
 	}
 
+	remaining := req.TLS.PeerCertificates[0].NotAfter.Sub(time.Now())
+	clientCertificateExpirationGauge.WithLabelValues(req.TLS.PeerCertificates[0].Subject.CommonName).Set(float64(remaining / time.Second))
```

Comment:
We want something that can be set to have alerts for monitoring in production. I'll see what can be done with node status or events.